packages/x-charts/src/ChartsOverlay/ChartsOverlay.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { SxProps, Theme } from '@mui/material/styles';
+import { ChartsLoadingOverlay } from './ChartsLoadingOverlay';
+import { useSeries } from '../hooks/useSeries';
+import { SeriesId } from '../models/seriesType/common';
+import { ChartsNoDataOverlay } from './ChartsNoDataOverlay';
+
+export function useNoData() {
+  const seriesPerType = useSeries();
+
+  return Object.values(seriesPerType).every((seriesOfGivenType) => {
+    if (!seriesOfGivenType) {
+      return true;
+    }
+    const { series, seriesOrder } = seriesOfGivenType;
+
+    return seriesOrder.every((seriesId: SeriesId) => series[seriesId].data.length === 0);
+  });
+}
+
+export type CommonOverlayProps = React.SVGAttributes<SVGTextElement> & {
+  /**
+   * The message displayed by the overlay.
+   */
+  message?: string;
+  sx?: SxProps<Theme>;
+};
+
+export interface ChartsOverlaySlots {
+  /**
+   * Loading overlay component rendered when the chart is in a loading state.
+   * @default ChartsLoadingOverlay
+   */
+  loadingOverlay?: React.ElementType<CommonOverlayProps>;
+  /**
+   * No data overlay component rendered when the chart has no data to display.
+   * @default ChartsNoDataOverlay
+   */
+  noDataOverlay?: React.ElementType<CommonOverlayProps>;
```

Comment:
What do you think about `emptyDataset` vs `noData`. I find "no data" a bit strange, but I'm not sure about "empty dataset" either 🤔

Reply:
The issue with `emptyDataset` is that `dataset` is a props, and the overlay is more general than this props.

It comes from the `noRowsOverlay` of the data grid
https://mui.com/x/react-data-grid/components/#no-rows-overlay